### PR TITLE
Vendor in latest containers/storage and vendor/github.com/sirupsen

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

Need to wait for containers/image.

### CODE REVIEW

**Comment by rhatdan** [comment]
@mrunalp FYI

**Comment by nalind** [comment]
Please note the switchover to using `golang.org/x/crypto/ssh/terminal` instead of functions that were dropped from the package in the commit log.

Looks like this will need https://github.com/containers/image/pull/319 to be merged first.

**Comment by rhatdan** [comment]
@nalind I think you need to take over this patch at this point.  I was trying to do it but got lost in the twisty little passages, around updating version of moby and docker packages.  Is there stuff we can get rid of to make this easier?

**Comment by nalind** [comment]
@rhatdan I'm not sure that this can be completed currently - the master branch of `github.com/openshift/imagebuilder` with the master branch of `github.com/fsouza/go-dockerclient` pulls in a pinned `github.com/docker/docker/opts` and `github.com/docker/docker/daemon/cluster/convert` which needs `github.com/docker/swarmkit`, which even on its master branch still uses the `logrus` package by its old name.

Cherry-picking https://github.com/nalind/buildah-1/commit/470753cfa1b9751630dd4599434cf179e37acfbe onto your branch will get you as far as I have, if you want to keep at it.

**Comment by rhatdan** [comment]
Is there code that we can eliminate or merge into buildah directly?  This seems to be a big mess of packages.

**Comment by nalind** [comment]
We use `imagebuilder` to drive `build-using-dockerfile`.  After that the details aren't really something we can change here.

File: vendor.conf [hunk 1]
```diff
@@ -1,13 +1,14 @@
 github.com/BurntSushi/toml master
 github.com/Nvveen/Gotty master
 github.com/blang/semver master
-github.com/containers/image 106607808da3cff168be56821e994611c919d283
-github.com/containers/storage 5d8c2f87387fa5be9fa526ae39fbd79b8bdf27be
+github.com/containers/image 063852766c3e82ec8359ce5f6612e056f3efaa76
+github.com/containers/storage 43c477703fe73129e2186cf730601c52d309c3ef
 github.com/docker/distribution master
-github.com/docker/docker 0f9ec7e47072b0c2e954b5b821bde5c1fe81bfa7
+github.com/docker/docker master
```
Comment (nalind) [inline vendor.conf]: Try pinning this to revision 30eb4d8cdc422b023d5f11f29a82ecb73554183b, which is the version that `containers/image` lists in its `vendor.conf`.

**Comment by rhatdan** [comment]
I have dropped the vendored docker content to match containers/image.  Will see if test pass.

**Comment by rhatdan** [comment]
@nalind passes tests, but you wanted this to pass more extensive tests correct?

**Comment by nalind** [comment]
Yeah, I wanted this to go through the tests in #282, as much to give them a workout as anything.

**Comment by rhatdan** [comment]
Testing it now.

**Comment by rhatdan** [comment]
@nalind Tests pass PTAL  Especially at the buildah code.

**Comment by nalind** [comment]
I'd make a change to `add.go` as above, otherwise LGTM.

**Comment by nalind** [comment]
The merge commit that Red Hat CI is testing looks a bit off.  Can you rebase, if necessary, and repush?

**Comment by rhatdan** [comment]
@nalind passes.

File: cmd/buildah/common.go [hunk 2]
```diff
@@ -183,7 +188,7 @@ func validateFlags(c *cli.Context, flags []cli.Flag) error {
 				f := flag.(cli.StringFlag)
 				name := strings.Split(f.Name, ",")
 				val := c.String(name[0])
-				if ok, _ := regexp.MatchString("^-.+", val); ok {
+				if ok := re.MatchString(val); ok {
 					return errors.Errorf("option --%s requires a value", name[0])
 				}
 			}
```
Comment (nalind) [inline cmd/buildah/common.go]: This pair of closing braces isn't required Go syntax.  Not a blocker, though.

**Comment by rhatdan** [comment]
Ok I am merging.

**Pull request merged**